### PR TITLE
Refresh Python usage documentation

### PR DESCRIPTION
**Pull request by wjones127**
# Description

I rewrote most of the Python usage documentation.

 * Reflowed the usage documentation to start with loading, then look at log introspection, and finally querying tables. At the end I created a placeholder for writing, noting that it's not yet supported.
 * Added info about supported backends and data catalogs.
 * Added more examples and guidance on how to query Delta tables. Note: I'd like to include DataFusion here, but from what I've seen that's not yet possible, right?

By rewriting this I broke any links to sections *within* the page, but not to the page itself.

# Related Issue(s)

None

# Documentation

### CODE REVIEW

**Comment by matthewmturner** [comment]
hi @wjones127 i was interested in this from a datafusion angle as well and added example to the rust documentation for querying here #519.

i had some issues with this though and had to point to specific git commits to get it to work.  i havent tried since then though so not sure if the issue will still persist.  im less familiar with the python bindings, so take this with a grain of salt, but i believe the general structures are in place to replicate how we query in rust with the datafusion python bindings.

on a separate but somewhat related note, i am working on adding s3 support to datafusion (https://github.com/datafusion-contrib/datafusion-objectstore-s3).  my loose understanding of delta lake is that its often cloud based, so getting s3 support added to datafusion should make querying it easier.  i had actually previously tried querying deltalake on s3 and that was how i found out that datafusion didnt support it which started me on that path.

hope this helps!

**Review by houqp — DISMISSED** [review]
LGTM, thanks for the detailed write up @wjones127 !

**Comment by houqp** [comment]
FYI @fvaleye @zijie0

**Review by fvaleye — DISMISSED** [review]
Thanks @wjones127 

LGTM! There is a minor error in the CI before merging the PR.

File: python/docs/source/usage.rst [hunk 1]
```diff
@@ -12,99 +14,310 @@ Resolve partitions for current version of the DeltaTable
...
+To load the current version, use the constructor:
+
+.. code-block:: python
+
+    >>> dt = DeltaTable("../rust/tests/data/delta-0.2.0")
+
+Alternatively, if you have a data catalog you can load it by reference to a 
+database and table name. Currently only AWS Glue is supported.
+
+.. TODO: auth to data catalog?
```
Comment (fvaleye) [inline python/docs/source/usage.rst:34]: FYI #522 explains the requirements of the Data Catalog integration.

Comment (wjones127) [reply python/docs/source/usage.rst]: Thanks!

**Review by fvaleye — APPROVED** [review]

**Pull request merged**